emp/migrate_to_io2.py

```diff
@@ -0,0 +1,138 @@
+import threading
+
+AWS_REGION_ID=''
+AWS_ACCESS_KEY_ID=''
+AWS_SECRET_ACCESS_KEY=''
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='[%(asctime)s] %(levelname)s - %(message)s',
+)
+logger = logging.getLogger('PF9')
+
+class MigrateVolume:
+    def __init__(self):
+        self.ec2_client = boto3.client('ec2', aws_access_key_id=AWS_ACCESS_KEY_ID, aws_secret_access_key=AWS_SECRET_ACCESS_KEY, region_name=AWS_REGION_ID)
+        self.polltime = 30
+
+    def calculate_iops(self, volume_size, iops_per_gb):
+        iops = min(math.ceil(volume_size * iops_per_gb), 64000)
+        return iops
+    
+    def check_modification_status(self, volume_id):
+        logger.info("Fetching modfication status for volume [%s].", volume_id)
+        try:
+            response = self.ec2_client.describe_volumes_modifications(VolumeIds=[volume_id])
+        except botocore.exceptions.ClientError as err:
+            if err.response["Error"]["Code"] == "InvalidVolumeModification.NotFound":
+                logger.warning(err.response["Error"]["Message"])
+                return
+            raise err
+
+        if len(response["VolumesModifications"]) < 1:
+            logger.warning("Volume with id (%s) Not Found", volume_id)
+            return
+
+        while response["VolumesModifications"][0]["ModificationState"] != "completed" and response["VolumesModifications"][0]["ModificationState"] != "failed":
+            time.sleep(self.polltime)
+            response = self.ec2_client.describe_volumes_modifications(VolumeIds=[volume_id])
+            logger.info("Polling volume(%s) status: %s, progress: %s", volume_id, response["VolumesModifications"][0]["ModificationState"], response["VolumesModifications"][0]["Progress"])
+
+        timetaken = response["VolumesModifications"][0]["EndTime"] - response["VolumesModifications"][0]["StartTime"]
+        logger.info("Volume modified: %s: %s. Total timetaken: %s", volume_id, response["VolumesModifications"][0]["ModificationState"], timetaken)
+
+    def modify_volume_to_io2(self, volume_id, iops_per_gb):
+        logger.info('Modifying volume %s to io2', volume_id)
+
+        try:
+            # Get volume info.
+            response = self.ec2_client.describe_volumes(VolumeIds=[volume_id])
+            
+            if len(response["Volumes"]) < 1:
+                logger.warning("Volume with id (%s) Not Found", volume_id)
+                return
+            volume_info = response['Volumes'][0]
+
+            # Calculate iops based on the volume size and IOPS/GB
+            iops = self.calculate_iops(volume_info['Size'], iops_per_gb)
+
+            logger.info('Modifying volume %s to io2 with the following parameters:- IOPS/GB: %s, IOPS: %s', volume_id, iops_per_gb, iops)
+            # Modify the volume to io2 and set the IOPS and throughput.
+            self.ec2_client.modify_volume(
+                VolumeId=volume_id,
+                VolumeType='io2',
+                Iops=iops
+            )
```

Comment:
Add `MultiAttachEnabled=True` here. We want to modify volume to io2 and make them multi-attach enabled.

Reply:
done